### PR TITLE
Zone namespace support based on terarkdb dev 1.4 version

### PR DESCRIPTION
**Pull request by Yuanliang-Wang**
At present, bytedance has encountered various problems of insufficient performance of the SSD disk itself due to the unfriendly IO pattern. Based on the more friendly IO pattern support brought by Zone Namespace to the KV database, we have supported it on terarkdb Zone Namespace to give full play to the capabilities of the disk itself

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ BASE=$PWD
 OUTPUT=output
 mkdir -p $OUTPUT
 
-if [ "$WITH_TESTS" == "1" ];then
+if [ "$WITH_TESTS" = "1" ];then
```
Comment (mm304321141) [inline build.sh:10]: 这个 == 改成 = 是符合预期的么？

File: README.md [hunk 2]
```diff
@@ -7,7 +93,7 @@ drawbacks.
 
 **NOTES**
 - TerarkDB was only tested and production ready under Linux platform
-- Language bindings except C/C++ are not fully tested yet.
+- Language bindings except C/C are not fully tested yet.
```
Comment (mm304321141) [inline README.md:96]: 这里的 C++ 被改成 C，包括后面几个，是符合预期的修改么？

File: build.sh [hunk 3]
```diff
@@ -26,5 +26,6 @@ fi
 
 git submodule update --init --recursive
 
-cd $BASE/$OUTPUT && cmake ../ -DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=${WITH_TESTS} -DWITH_TOOLS=ON -DWITH_TERARK_ZIP=ON
+cd $BASE/$OUTPUT && cmake ../ -DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=${WITH_TESTS} -DWITH_TOOLS=ON
```
Comment (mm304321141) [inline build.sh:29]: DWITH_TERARK_ZIP 关掉的原因是什么？

File: README.md (hunk 2 above)
Comment (levisonchen) [reply README.md:96]: 这应该是手误。

File: build.sh (hunk 3 above)
Comment (levisonchen) [reply build.sh:29]: DWITH_TERARK_ZIP 的默认值是 ON，这里只是去掉多余的部分。

File: build.sh (hunk 1 above)
Comment (Yuanliang-Wang) [reply build.sh:10]: 符合，shell脚本，只用一个=号判断就可以

File: README.md (hunk 2 above)
Comment (Yuanliang-Wang) [reply README.md:96]: accept，我恢复回来

**Comment by Yuanliang-Wang** [comment]
There was a problem with the previous submission, and another amendment

**Pull request closed without merging**